### PR TITLE
Rspecs to Book and User models

### PR DESCRIPTION
**Pull request by aleochoam**
### Summary
- Added the rspec-rails gem
- Added the faker gem to generate random data
- Added the initial specs for User and Book models

### Trello card
https://trello.com/c/PmR1vuE1/21-crear-tests-para-el-modelo-de-book

### CODE REVIEW

**Comment by aleochoam** [comment]
This should not be closed

File: spec/models/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+require 'faker'
+require 'rails_helper'
+
+describe Book do
```
Comment (Gmarn) [inline spec/models/book_spec.rb]: Here I will prefer you use FactoryBot to generate Book model attributes. I know this is as the example you have on the card but is a better way to work with factories

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
+require 'faker'
+require 'rails_helper'
+
+describe User do
```
Comment (Gmarn) [inline spec/models/user_spec.rb]: Thanks to testing User model but card ask only to test Book model

Comment (Gmarn) [reply spec/models/user_spec.rb]: You will surely refactor test in this file, so ;)

Comment (aleochoam) [reply spec/models/user_spec.rb]: I will keep the test just in case

File: spec/factories/book_factory.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  factory :book do
+    genre { Faker::Book.genre }
+    author { Faker::Book.author }
+    image { Faker::Internet.url }
+    title { Faker::Book.title }
+    publisher { Faker::Book.publisher }
+    year { Faker::Date.backward(1000).year }
+  end
+
+  factory :book_without_publisher, class: Book do
```
Comment (Gmarn) [inline spec/factories/book_factory.rb]: This is a good idea but there are better ways to do that.

1. is to use the book factory and call it as that: `bulid(:books, publisher: '')`
2. use `trait`

Here better to use 1st option.

File: spec/models/book_spec.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    subject(:book) do
+      build(:book)
+    end
+
+    it 'is valid' do
+      expect(subject.valid?).to eq(true)
+    end
+  end
+
+  context 'without all fields' do
```
Comment (Gmarn) [inline spec/models/book_spec.rb]: Here you are only testing if `publisher` is required, but you need also to test other fields.
And remember a unit test, test one feature by one feature.

File: spec/models/book_spec.rb [hunk 5]
```diff
@@ -0,0 +1,29 @@
+require 'faker'
+require 'rails_helper'
+
+describe Book do
+  context 'with all fields' do
+    subject(:book) do
```
Comment (Gmarn) [inline spec/models/book_spec.rb]: `subject` should not be in `context` by convention a `subject` is link to a `describe`

File: spec/models/book_spec.rb [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+      build(:book)
+    end
+
+    it 'is valid' do
+      expect(subject.valid?).to eq(true)
+    end
+  end
+
+  context 'without all fields' do
+    subject(:book) do
```
Comment (Gmarn) [inline spec/models/book_spec.rb]: same

File: .travis.yml [hunk 7]
```diff
@@ -1,3 +1,37 @@
...
+  - psql -c "DROP DATABASE IF EXISTS \"WBooksAPI_test\";" -U postgres
+  - psql -c "CREATE DATABASE \"WBooksAPI_test\" ENCODING 'utf8';" -U postgres
+  - bundle exec rake db:migrate
+
+script:
+  - bundle exec rspec spec -fd
+  - bundle exec rubocop app -R --format simple
+
+cache:
+  bundler: true
```
Comment (Gmarn) [inline .travis.yml]: Becareful you miss the end of line here. Better if you configure your IDE to highlight this.

File: spec/factories/user_factory.rb [hunk 8]
```diff
@@ -0,0 +1,25 @@
+FactoryBot.define do
+  factory :user do
+    first_name { Faker::Name.first_name }
+    last_name  { Faker::Name.last_name }
+    email      { Faker::Internet.email }
+    password   { 'password' }
+    password_confirmation   { 'password' }
+  end
+
+  factory :user_without_firstname, class: User do
```
Comment (Gmarn) [inline spec/factories/user_factory.rb]: same comment here from book

File: spec/models/user_spec.rb [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+      build(:user)
+    end
+
+    it 'is valid' do
+      expect(subject.valid?).to eq(true)
+    end
+  end
+
+  context 'without first name' do
+    subject(:user) do
```
Comment (Gmarn) [inline spec/models/user_spec.rb]: Same comments here than book test

File: spec/factories/book_factory.rb (hunk 3 above)
Comment (aleochoam) [reply spec/factories/book_factory.rb]: I did not knew you could do that! Great

File: spec/models/book_spec.rb [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+
+  context 'with all fields' do
+    let(:book) { build(:book) }
+
+    it 'is valid' do
+      expect(subject.valid?).to eq(true)
+    end
+  end
+
+  context 'without publisher' do
```
Comment (Gmarn) [inline spec/models/book_spec.rb]: Nice, here you are testing all fields, can you think about something DRYier ?

File: spec/models/book_spec.rb [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+describe Book do
+  subject(:book) do
+    build(:book)
+  end
+
+  it 'is valid with all atributes' do
+    is_expected.to be_valid
+  end
+
+  it 'is not valid without a publisher' do
```
Comment (Gmarn) [inline spec/models/book_spec.rb]: Not on this way. Test was better written before. A tip:
```ruby
%i[genre author image title publisher year].each do |field|
```

(there is an other way with `share_example` and  `include_example` but it is only for information, just keep it in mind for the future)

File: spec/models/user_spec.rb [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+describe User do
+  subject(:user) do
+    build(:user)
+  end
+
+  it 'is valid with all fields' do
+    is_expected.to be_valid
+  end
+
+  it 'is not valid without a first_name' do
```
Comment (Gmarn) [inline spec/models/user_spec.rb]: same

**Pull request merged**